lib/string_reverse.rb

```diff
@@ -1,4 +1,30 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  # my_string << "not implemented"
+  if my_string == nil
+    return nil
+  end
+
+  i = 0
+  j = my_string.length - 1
+  while i < j
+    first_letter = my_string[i]
+    last_letter = my_string[j]

```

Comment:
you can achieve the same effect by only assigning one temporary variable (you don't need two). BUT it's so much easier to follow with the two.

Reply:
Agreed, you should be able to swap with just one temporary variable like so:
```ruby
while i < j
   temp = my_string[i]
   my_string[i] = my_string[j]
   my_string[j] = temp
   i += 1
   j -= 1
end
```